src/frontend/hooks/google-auth.js

```diff
@@ -0,0 +1,76 @@
+import { useCallback } from 'react';
+import { useGoogleLogin } from 'react-google-login';
+import * as userActions from '../state/user/actions';
+import { useDispatch } from 'react-redux';
+import api from '../api';
+import { useGoogleLogout } from 'react-google-login';
+import Cookies from 'js-cookie';
+
+// TODO: store token in localStorage.
+const useGoogleAuth = (onAuthComplete) => {
+  const dispatch = useDispatch();
+  const onSuccess = useCallback(
+    (response) => {
+      // https://github.com/anthonyjgrove/react-google-login/blob/7db5b9686a70ded6b090a9c01906ca978b00a54d/index.d.ts#L29
+      const { tokenId, profileObj, accessToken } = response;
+      console.log('Begin auth');
+      api.google
+        .checkToken(tokenId, accessToken)
+        .then((checkTokenResponse) => {
+          if (checkTokenResponse.status === 200) {
+            const { userId, groupIds, accessToken } = checkTokenResponse.data;
+            onAuthComplete(null, {
+              userId,
+              tokenId,
+              groupIds,
+              accessToken,
+            });
+          }
+        })
+        .catch((err) => onAuthComplete(err));
```

Comment:
What happens when status !== 200. Does it just hop into the catch block and get handled from there ?

Reply:
Good point, we should throw an error if status !== 200